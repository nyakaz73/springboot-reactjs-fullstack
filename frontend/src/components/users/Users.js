import React from 'react'
import UserInfo from '../user-info/UserInfo';
import PropTypes from "prop-types";
const Users = ({ users, removeUser }) => {
    return (
        <div style={(users !== undefined || users.length !== 0) ? container : null}>
            {
                users.map(user => (
                    <UserInfo user={user} key={user.id} removeUser={removeUser} />
                ))
            }
        </div>
    )
}


const container = {
    border: '2px solid #544073',
    margin: '50px',
    padding: '10px'
}

Users.propTypes = {
    users: PropTypes.array,
    removeUser: PropTypes.func.isRequired
}
export default Users;
