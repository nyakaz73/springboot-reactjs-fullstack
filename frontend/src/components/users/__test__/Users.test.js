import React from "react";
import ReactDOM from "react-dom";
import { act, fireEvent, render } from "@testing-library/react";
import Users from "../Users";
import renderer from "react-test-renderer"

let container;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    document.body.removeChild(container);
    container = null;
});


it("Renders <Users/> without crashing", () => {
    const removeUser = jest.fn()
    const users = [
        {
            id: "1",
            name: "John",
            surname: "Marcus",
            username: "johnny",
            email: "john@gmail.com",
            password: "some-strong-password",
        },
        {
            id: "2",
            name: "Peter",
            surname: "Rubel",
            username: "peter",
            email: "peter@gmail.com",
            password: "some-strong-password2",
        },
        {
            id: "3",
            name: "Daniel",
            surname: "James",
            username: "daniel",
            email: "daniel@gmail.com",
            password: "some-strong-password3",
        },
    ];
    // Simulate an async call
    global.fetch = jest.fn().mockImplementation(() =>
        Promise.resolve(users)
    );
    act(() => {
        ReactDOM.render(<Users users={users} removeUser={removeUser} />, container);
    });
    expect(container.textContent).toContain(users[0].email);
    expect(container.textContent).toContain(users[0].name);
    expect(container.textContent).toContain(users[0].surname);
    expect(container.textContent).toContain(users[1].email);
    expect(container.textContent).toContain(users[2].email);
})

it("matches snapshot", () => {
    const removeUser = jest.fn()
    const users = []
    const tree = renderer.create(<Users users={users} removeUser={removeUser} />).toJSON();
    expect(tree).toMatchSnapshot();

})