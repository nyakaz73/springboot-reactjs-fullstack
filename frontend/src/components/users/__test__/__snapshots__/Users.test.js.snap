// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  style={
    Object {
      "border": "2px solid #544073",
      "margin": "50px",
      "padding": "10px",
    }
  }
/>
`;
